CIPs/CIP-19/CIP-19.md

```diff
@@ -0,0 +1,173 @@
+
+| Property           | Description                    | Value                                                        | Max Size | Required | Example                      |
+| ------------------ | ------------------------------ | ------------------------------------------------------------ | -------- | -------- | ---------------------------- |
+| `name`             | a name                         | string                                                       | 150 char | optional | Mary Smith                   |
+| `image`            | an image                       | IPFS CID                                                     |          | optional |                              |
+| `description`      | a short description            | string                                                       | 420 char | optional | This is my cool description. |
+| `emoji`            | an emoji                       | unicode                                                      | 2 char   | optional | 🔢                            |
+| `background`       | a background image (3:1 ratio) | IPFS CID                                                     |          | optional |                              |
+| `birthDate`        | a date of birth                | [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)           | 10 char  | optional | 1990-04-24                   |
+| `url`              | a url                          | string                                                       | 240 char | optional | http://ceramic.network       |
+| `gender`           | a gender                       | string                                                       | 42 char  | optional | female                       |
+| `residenceCity`    | a city of residence            | string                                                       | 140 char | optional | Berlin                       |
+| `residenceCountry` | a country of residence         | [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) | 2 char   | optional | DE                           |
+| `nationalities`    | nationalities                  | array of [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) values | 2 char   | optional | CN                           |
+| `affiliations`     | affiliations                   | array of strings                                             | 240 char | optional | Ceramic Ecosystem Alliance   |
+
+**Deployment:** `<reference-schema-DocID>`
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "BasicProfile",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string",
+      "maxLength": 150
+    },
+    "image": {
+      "type": "string"
+    },
```

Comment:
@PaulLeCam thoughts about changing this to be a string that starts with `ipfs://`? Same thing for background.

Reply:
I think we should make the type more strict but limiting to `ipfs://` might limit use cases. We'd likely also need to include `ipns://`, but what about other protocols and possible browser extensions (ex `web+ipfs://`)?
Maybe instead we can restrict it to the [`uri` format](https://json-schema.org/understanding-json-schema/reference/string.html#uri-template)?

Reply:
I'd rather be restrictive here so that developers know what to expect. This will in most cases need to be handled by developers in their app, so the fewer options the better imo.

Reply:
Yes I agree it's a good thing to be more restrictive here, but I feel IPFS only is very restrictive and might limit usage for projects that would otherwise use this definition. If we want to limit to a single protocol I think `https://` would be an more convenient one for most projects, at least in the short term.

Reply:
Don't agree at all tbh. The profiles should be trustless. Using HTTPS opens up attack vectors which I'm definitely not comfortable with. Ipfs is extremely descriptive because of the content addressing. I think immutability is what we should aim for here. This is critical for a lot of initial use cases of IDX where trust is extremely important, e.g. decentralized governance.

Reply:
Yes I totally agree this is the "web3 ideal" and we should definitely aim for this, but as we're trying to figure out the "web 2.5 bridge" we should also consider what this means in terms of UX and DX.
Currently is that it's not really possible to be fully decentralized, we're still using gateways for IPFS, hosting on AWS, etc. so we're already making these trade-offs at a different layer. By forcing to use IPFS here, I'm concerned we're "forcing" users and dev to the opt for the most secure option, even though we're still relying on centralized options ourselves.
I think there is a risk we will limit the potential use case for the basic profile definition and that it could be used as a "universal default" if we make it too constraining. Developers could simply use an alternative schema allowing HTTPS that would "compete" with our basic profile, and split the ecosystem rather than helping it converge to a single go-to option.
Maybe we could discuss this with other people in the ecosystem to make sure we base these kind of decisions on current product needs rather than decentralizations ideals?

Reply:
Most of our initial customers care deeply about this and if we start getting competing basic profiles in IDX that's actually great because we get more usage of IDX.

Reply:
Using gateways for ipfs still provide much more guarantees because of content adressability, and we also have the option to upgrade very easily to more secure alternatives (once we start running ipfs and ceramic in the browser).